test/util/random.cpp

```diff
@@ -73,16 +73,18 @@ BOOST_AUTO_TEST_CASE(TestWalkerDiscreteDistributionSimple) {
 
   size_t iter = 1000000; // large enough
   for (size_t i = 0; i < iter; i++) {
-    res[s()]++;
-  }
-
-  std::vector<double> res_p;
-  for (size_t f: res) {
-    res_p.push_back(static_cast<double>(f) / static_cast<double>(iter));
+    size_t index = s();
+    BOOST_CHECK(index < len);
+    res[index]++;
   }
 
+  constexpr double Z = 1.96; // alpha=0.05, Z_{0.025}
```

Comment:
I understand that, in this part, the statistical test against binomial distributions is performed `len` times.
Probably it would be helpful if you leave a comment about it.
Also, as a CI test, I prefer much smaller p-values because there is a 5% chance of failure.